src/main/java/src/main/java/com/sorts/CountingSort.java

```diff
@@ -0,0 +1,52 @@
+package src.main.java.com.sorts;
+
+public class CountingSort {
+
+    /**
+     * This method sorts the array using counting sort technique
+     *
+     * @param arr The array to be sorted
+     * @return arr Sorted array
+     */
+    public Integer[] sort(Integer[] arr) {
+
+        // Finding the maximum element in the array
+        int max = arr[0];
+        for (int i = 1; i < arr.length; i++) {
+            max = Math.max(max, arr[i]);
+        }
+
+        // Creating the count array - This method will store the count of each element in the unsorted array
+        int[] count = new int[max + 1];
+
+        // This loop will store the count of each element in the array
+        for (int i = 0; i < max; i++) {
+
+            count[arr[i]]++;
```

Comment:
what will happen if arr[i] less than zero ?

Reply:
@yanglbme good find! I have fixed the issue by adding the offset for the negative values.

Reply:
@ani03sha 👍